src/Game/Game.elm

```diff
@@ -115,7 +115,10 @@ update msg model =
             ( { model | currentScreen = InventoryScreen }, Cmd.none )
 
         InvMsg msg ->
-            ( Inventory.update msg model, Cmd.none )
+            if msg == ExitMsg then
+                ( { model | currentScreen = MapScreen }, Cmd.none )
```

Comment:
Over here we're dipping into the Inventory's model, which is usually a good sign that something might be out of place (bit of a code smell in Elm). Can you think of a better way to do this? I feel like a command might be the right way to go.

When the Inventory's model is initialised in the Game, can we pass an exit command from the game into it? That way the Inventory (inner context) gets to know nothing about the outer context (the Game), and the inner context doesn't start to "bleed up" into the outer. Also, the outer context doesn't start doing things that maybe should be the responsibility of the inner context (such as triggering the command to exit).

This is a very tricky pattern to solve cleanly. Elm shows it for what it is: a difficult thing to solve in a way that keeps two parts cleanly separated.

Love your enthusiasm! :)